modules/openideal_webhooks/src/Plugin/RulesAction/WebhookAction.php

```diff
@@ -0,0 +1,196 @@
+<?php
+
+namespace Drupal\openideal_webhooks\Plugin\RulesAction;
+
+use Drupal\Component\Serialization\Json;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Drupal\message\MessageInterface;
+use Drupal\rules\Core\RulesActionBase;
+use Drupal\webhooks\Webhook;
+use Drupal\webhooks\WebhooksService;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\Serializer\SerializerInterface;
+
+/**
+ * Provides a Webhook action.
+ *
+ * @RulesAction(
+ *   id = "openideal_webhook",
+ *   deriver = "Drupal\openideal_webhooks\Plugin\RulesAction\WebhookActionDeriver",
+ * )
+ */
+class WebhookAction extends RulesActionBase implements ContainerFactoryPluginInterface {
+
+  use StringTranslationTrait;
+
+  /**
+   * Outgoing webhook types.
+   */
+  const SLACK = 'slack';
+  const OTHER = 'other';
+
+  /**
+   * Event dispatcher.
+   *
+   * @var \Symfony\Component\EventDispatcher\EventDispatcherInterface
+   */
+  protected $webhookService;
+
+  /**
+   * Entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Serializer.
+   *
+   * @var \Symfony\Component\Serializer\SerializerInterface
+   */
+  protected $serializer;
+
+  /**
+   * {@inheritDoc}
+   */
+  public function __construct(
+    array $configuration,
+    $plugin_id,
+    $plugin_definition,
+    WebhooksService $webhookService,
+    EntityTypeManagerInterface $entityTypeManager,
+    SerializerInterface $serializer
+  ) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->webhookService = $webhookService;
+    $this->entityTypeManager = $entityTypeManager;
+    $this->serializer = $serializer;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('webhooks.service'),
+      $container->get('entity_type.manager'),
+      $container->get('serializer')
+    );
+  }
+
+  /**
+   * Dispatch webhooks event.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   Entity from fetch data.
+   * @param string $event
+   *   Event id.
+   */
+  protected function doExecute(EntityInterface $entity, $event) {
+    // Remove default part of id.
+    $id = preg_replace('/[a-z0-9_]+:/', '', $this->getPluginId());
```

Comment:
As far as I understand, `getPluginId()` method will always return either `slack` or `default`.
Does it make sense to make `preg_replace` here?

Reply:
No, it will return plugin_base_id:id here is example https://github.com/linnovate/openideal/pull/162/files#diff-69414b4af13f88753f14586a872d2ae9db0150c64d743e8ef790baf04739d7e6R36